aiosmtpd/tests/test_smtp.py

```diff
@@ -202,1404 +211,1420 @@ class SleepingHeloHandler:
-            self.assertEqual(response, b"UGFzc3dvcmQA")
-            code, response = client.docmd('=')
-            assert_auth_invalid(self, code, response)
-
-
-class TestSMTPAuth(unittest.TestCase):
-    def setUp(self):
-        self.handler = PeekerHandler()
-        controller = DecodingControllerPeekAuth(
-            self.handler, server_kwargs={"auth_exclude_mechanism": ["DONT"]}
+        _, responses = transport_resp
+        assert responses[5] == b"250 OK\r\n"
+        assert len(handler.box) == 1
+        assert handler.box[0].content == b""
+
+
+# Because decoding_authnotls_controller has a scope of "function", this fixture will
+# be automagically started and teardown-ed on each test case func
+@pytest.mark.usefixtures("decoding_authnotls_controller")
+class TestSMTP(_CommonMethods):
+    valid_mailfrom_addresses = [
+        # no space between colon and address
+        "anne@example.com",
+        "<anne@example.com>",
+        # one space between colon and address
+        " anne@example.com",
+        " <anne@example.com>",
+        # multiple spaces between colon and address
+        "  anne@example.com",
+        "  <anne@example.com>",
+        # non alphanums in local part
+        "anne.arthur@example.com",
+        "anne+promo@example.com",
+        "anne-arthur@example.com",
+        "anne_arthur@example.com",
+        "_@example.com",
+        # IP address in domain part
+        "anne@127.0.0.1",
+        "anne@[127.0.0.1]",
+        "anne@[IPv6:2001:db8::1]",
+        "anne@[IPv6::1]",
+        # email with comments -- obsolete, but still valid
+        "anne(comment)@example.com",
+        "(comment)anne@example.com",
+        "anne@example.com(comment)",
+        "anne@machine(comment).  example",  # RFC5322 § A.6.3
+        # source route -- RFC5321 § 4.1.2 "MUST BE accepted"
+        "<@example.org:anne@example.com>",
+        "<@example.net,@example.org:anne@example.com>",
+        # strange -- but valid -- addresses
+        "anne@mail",
+        '""@example.com',
+        '<""@example.com>',
+        '" "@example.com',
+        '"anne..arthur"@example.com',
+        "mailhost!anne@example.com",
+        "anne%example.org@example.com",
+        'much."more\\ unusual"@example.com',
+        'very."(),:;<>[]".VERY."very@\\ "very.unusual@strange.example.com',
+        # more from RFC3696 § 3
+        # 'Abc\\@def@example.com', -- get_addr_spec does not support this
+        "Fred\\ Bloggs@example.com",
+        "Joe.\\\\Blow@example.com",
+        '"Abc@def"@example.com',
+        '"Fred Bloggs"@example.com',
+        "customer/department=shipping@example.com",
+        "$A12345@example.com",
+        "!def!xyz%abc@example.com",
+    ]
+
+    valid_rcptto_addresses = valid_mailfrom_addresses + [
+        # Postmaster -- RFC5321 § 4.1.1.3
+        "<Postmaster>",
+    ]
+
+    invalid_email_addresses = [
+        "<@example.com>",  # no local part
+        "a" * 65 + "@example.com",  # local-part > 64 chars
+    ]
```

Comment:
These are deemed invalid addresses by the email standards.

Oh yeah, for the "local-part > 64 chars" test I had to modify the `_getaddr` method in `SMTP`.